backend/oasst_backend/api/v1/text_labels.py

```diff
@@ -14,19 +14,23 @@
 @router.post("/", status_code=HTTP_204_NO_CONTENT)
 def label_text(
     *,
-    db: Session = Depends(deps.get_db),
+    # db: Session = Depends(deps.get_db),
     api_key: APIKey = Depends(deps.get_api_key),
     text_labels: protocol_schema.TextLabels,
 ) -> None:
     """
     Label a piece of text.
     """
-    api_client = deps.api_auth(api_key, db)
+
+    @managed_tx_function(CommitMode.COMMIT)
+    def store_text_labels(session: deps.Session, api_key):
+        api_client = deps.api_auth(api_key, session)
+        pr = PromptRepository(session, api_client, client_user=text_labels.user)
+        pr.store_text_labels(text_labels)
 
     try:
         logger.info(f"Labeling text {text_labels=}.")
-        pr = PromptRepository(db, api_client, client_user=text_labels.user)
-        pr.store_text_labels(text_labels)
+        store_text_labels(api_key=api_key)
 
     except Exception:
```

Comment:
```suggestion
    except OasstError:
        raise
    except Exception:
```

btw if you at it ... that would allow OasstErrors to get to the default handler...